### PR TITLE
Experimental Debian base image

### PR DESCRIPTION
**Pull request by alex-namely**
1. Use Debian base image
2. Use Bazel to build grpc from source codes

Build time ~20 minutes

### CODE REVIEW

**Comment by vtopc** [comment]
19m - nice time. LGTM

File: Dockerfile [hunk 1]
```diff
@@ -91,37 +88,39 @@ RUN curl -sSL https://github.com/grpc/grpc-web/releases/download/${grpc_web_vers
...
 RUN npm i -g ts-protoc-gen@0.12.0
 
-COPY --from=build /tmp/grpc-java/compiler/build/exe/java_plugin/protoc-gen-grpc-java /usr/local/bin/
 COPY --from=build /tmp/googleapis/google/ /opt/include/google
 COPY --from=build /tmp/api-common-protos/google/ /opt/include/google
+
+# Copy well known proto files
+COPY --from=build /tmp/grpc/bazel-grpc/external/com_google_protobuf/src/google/protobuf/ /opt/include/google/protobuf/
+# Copy protoc
+COPY --from=build /tmp/grpc/bazel-bin/external/com_google_protobuf/ /usr/local/bin/
```
Comment (vtopc) [inline Dockerfile:116]: If it's copying only one binary, shouldn't this be more explicit? e.g.:
```suggestion
COPY --from=build /tmp/grpc/bazel-bin/external/com_google_protobuf/protoc /usr/local/bin/
```

so if something goes wrong we would have an error during building this image, not in runtime.

File: Dockerfile [hunk 2]
```diff
@@ -91,37 +88,39 @@ RUN curl -sSL https://github.com/grpc/grpc-web/releases/download/${grpc_web_vers
...
+# Copy protoc default plugins
+COPY --from=build /tmp/grpc/bazel-bin/src/compiler/ /usr/local/bin/
+# Copy protoc java plugin
+COPY --from=build /tmp/grpc-java/bazel-bin/compiler/ /usr/local/bin/
+# Copy grpc_cli
+COPY --from=build /tmp/grpc/bazel-bin/test/cpp/util/ /usr/local/bin/
```
Comment (vtopc) [inline Dockerfile:122]: ... same here.

File: Dockerfile (hunk 1 above)
Comment (vtopc) [reply Dockerfile:116]: oh, OK, nvm

**Review by vtopc — APPROVED** [review]
@abe545, @ido-namely, any comments?

File: Dockerfile [hunk 3]
```diff
@@ -1,60 +1,57 @@
...
-RUN set -ex && apk --update --no-cache add \
-    protoc~=${proto_version} \
-    protobuf~=${proto_version} \
-    protobuf-dev~=${proto_version} \
-    --repository=http://dl-cdn.alpinelinux.org/alpine/edge/main
+RUN git clone -b v$grpc_version.x --recursive -j8 --depth 1 https://github.com/grpc/grpc && \ 
+    git clone -b v$grpc_java_version.x --recursive https://github.com/grpc/grpc-java.git && \
+    git clone -b v$grpc_version.x --recursive https://github.com/grpc/grpc-go.git && \
+    git clone https://github.com/googleapis/googleapis && \
+    git clone https://github.com/googleapis/api-common-protos
```
Comment (vtopc) [inline Dockerfile:33]: what about `--depth 1` for all of these?

Comment (alex-namely) [reply Dockerfile:33]: sounds good to me

**Review by abe545 — APPROVED** [review]

**Review by ido-namely — APPROVED** [review]

**Pull request merged**